pkg/precision/precision.go

```diff
@@ -0,0 +1,54 @@
+package precision
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/shopspring/decimal"
+)
+
+func ToSignificant(input decimal.Decimal, sigDigits, maxPrecision int32) decimal.Decimal {
```

Comment:
I don't think we need a maxPrecision here, we are going to use Validate before and if ok we call ToSignificant, do you agree @maxpushka @dimast-x

Reply:
In case we remove `maxPrecision`, we need to think of a mechanism that prevents overflows, that is exponent will be larger than `maxPrecision`. We have parts of code where price is multiplied by amount and this may result in such event.

Without `maxPrecision` param the package has not way to enforce entries 1 and 2 of Consequences section of YIP-0001. Imagine a case when backend returns a number that, in case it's piped back in, cannot pass verifications. This will lead to breaking the contract